tools/proc_init.py

```diff
@@ -0,0 +1,302 @@
+Invoke the Node Data here as well. it will serve the purpose. 
+'''
+def recvEdgeData (rank, dimensions, dtype): 
+    return recvNodeData (dimensions, dtype)
+
+'''
+This function is for receiving node features. 
+Note that the node features are indexed by the orig_node_type_id. 
+for the mag-dataset, only node_type = 3 has node features and others node types does not
+have any node features. 
+'''
+def recvNodeFeatures (rank, dtype): 
+    globalIds = recvNodeData( rank, 1, torch.int32 ) ## to receive the origTypeNodeIDs
+    node_features = recvNodeData( rank, 2, torch.float32 ) ## to receive actual node feature data
+    print( 'Rank: ', rank, ', Done receiving node feature data... ' )
+    return globalIds, node_features
+
+def recv_node_features_obj(rank, size): 
+    send_objs = [None for _ in range(size)]
+    recv_obj = [None]
+    dist.scatter_object_list(recv_obj, send_objs, src=0)
+
+    node_features = recv_obj[0]
+    return node_features
+
+
+def readInputFiles( rank, params, metis_partitions ): 
+
+    '''
+    Node data is structured as follows: 
+    <node_type> <weight1> <weight2> <weight3> <weight4> <orig_type_node_id> <attributes>
+    is converted to 
+    <node_type> <weight1> <weight2> <weight3> <weight4> <orig_type_node_id> <nid> <recv_proc>
+    '''
+    rcvProc_nodes_data = []
+    nodes_data = read_nodes_file( params.input_dir+'/'+params.nodes_file )
+    rcvProc_nodes_data = include_recv_proc_nodes( nodes_data, metis_partitions )
+    print( 'Rank: ', rank, ', Completed loading nodes data: ', rcvProc_nodes_data.shape )
+
+    rcvProc_edge_data = []
+    edges_data = read_edge_file( params.input_dir+'/'+params.edges_file )
+    rcvProc_edge_data = include_recv_proc_edges( edges_data, metis_partitions )
+    print( 'Rank: ', rank, ', Completed loading edges data: ', rcvProc_edge_data.shape )
+
+    node_features = []
+    node_features = read_node_features_file( params.input_dir+'/'+params.node_feats_file )
+    print( 'Rank: ', rank, ', Completed loading node features reading from file ', len(node_features))
+
+    edge_features = []
+    #edge_features = read_edge_features_file( params.input_dir+'/'+params.edge_feats_file )
+    #print( 'Rank: ', rank, ', Completed edge features reading from file ', len(edge_features) )
+
+    return rcvProc_nodes_data, node_features, rcvProc_edge_data, edge_features
+
+def run(rank, size, params):
+
+    metis_partitions = read_metis_partitions(params.input_dir+'/'+params.metis_partitions)
+    print( 'Rank: ', rank, ', Completed loading metis partitions: ', len(metis_partitions))
+
+    schema_map = read_json(params.input_dir+'/'+params.schema)
+    ntypes_map, ntypes = get_node_types(schema_map)
+
+    if rank == 0: 
+        node_data, node_features, edge_data, edge_features = readInputFiles( rank, params, metis_partitions )
+
+        # order node_data by node_type before extracting node features. 
+        # once this is ordered, node_features are automatically ordered and 
```

Comment:
i don't understand this command. why are `node_features` ordered automatically after `node_data` is ordered?